python/pyarrow/_dataset.pyx

```diff
@@ -748,7 +687,7 @@ cdef class DatasetFactory:
 
 cdef class Dataset:
```

Comment:
Move this a bit up, since the subclasses are defined above?

Reply:
Will do